nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,432 @@
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void setUpClient(final ProcessContext context)  {
+        WebClientServiceProvider standardWebClientServiceProvider = context.getProperty(WEB_CLIENT_SERVICE).asControllerService(WebClientServiceProvider.class);
+        RecordReaderFactory recordReaderFactory = context.getProperty(RECORD_READER_FACTORY).asControllerService(RecordReaderFactory.class);
+        RecordSetWriterFactory recordSetWriterFactory = context.getProperty(RECORD_WRITER_FACTORY).asControllerService(RecordSetWriterFactory.class);
+        WebClientService webClientService = standardWebClientServiceProvider.getWebClientService();
+        webClientAtomicReference.set(webClientService);
+        recordReaderFactoryAtomicReference.set(recordReaderFactory);
+        recordSetWriterFactoryAtomicReference.set(recordSetWriterFactory);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowfile = session.get();
+
+        if (skipExecution(context, flowfile)) {
+            return;
+        }
+
+        ComponentLog logger = getLogger();
+        FlowFile responseFlowFile = null;
+
+        try {
+            WebClientService webClientService = webClientAtomicReference.get();
+            URI uri = new URI(context.getProperty(REPORT_URL).evaluateAttributeExpressions(flowfile).getValue().trim());
+            long startNanos = System.nanoTime();
+            String authorization = createAuthorizationHeader(context, flowfile);
+
+            try(HttpResponseEntity httpResponseEntity = webClientService.get().uri(uri).header(HEADER_AUTHORIZATION, authorization).retrieve()) {
+                responseFlowFile = createResponseFlowFile(flowfile, session, context, httpResponseEntity);
+                long elapsedTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+                Map<String, String> commonAttributes = createCommonAttributes(uri, httpResponseEntity, elapsedTime);
+
+                if (flowfile != null) {
+                    flowfile = session.putAllAttributes(flowfile, decorateWithMimeAttribute(commonAttributes, httpResponseEntity));
+                }
+                if (responseFlowFile != null) {
+                    responseFlowFile = session.putAllAttributes(responseFlowFile, commonAttributes);
+                    if (flowfile != null) {
+                        session.getProvenanceReporter().fetch(responseFlowFile, uri.toString(), elapsedTime);
+                    } else {
+                        session.getProvenanceReporter().receive(responseFlowFile, uri.toString(), elapsedTime);
+                    }
```

Comment:
Recommend swapping the logic to use `==` instead of `!=`:
```suggestion
                    if (flowfile == null) {
                        session.getProvenanceReporter().receive(responseFlowFile, uri.toString(), elapsedTime);
                    } else {
                        session.getProvenanceReporter().fetch(responseFlowFile, uri.toString(), elapsedTime);
                    }
```

Reply:
done